pkg/controllers/status.go

```diff
@@ -20,10 +20,11 @@ import (
 
 // The default set of status change reasons.
 const (
-	ReasonAsExpected   = "AsExpected"
-	ReasonInitializing = "Initializing"
-	ReasonSyncing      = "SyncingResources"
-	ReasonSyncFailed   = "SyncingFailed"
+	ReasonAsExpected     = "AsExpected"
+	ReasonInitializing   = "Initializing"
+	ReasonSyncing        = "SyncingResources"
+	ReasonSyncFailed     = "SyncingFailed"
+	ReasonNotUpgradeable = "NotUpgradeable"
```

Comment:
nit: doesn't really matter for a one-off, but `Upgradeable=False` has the type/status already saying "not upgradeable".  Ideally `reason` adds some context like `MissingNutanixConfigMap` that's specific to the thing we're asking the admin to fix.

Reply:
thanks, i wasn't sure about this one. i'll make it more descriptive/specific.